src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -203,21 +218,93 @@ private void CheckServerState()
         }
     }
 
-    public async Task ShutdownAsync(string message = "Shutting down")
+    public async Task WaitForExitAsync()
```

Comment:
used by the AbstractInprocLanguageClient to ensure the server has fully shutdown before we restart (the client doesn't know when we've completed exit since its a notification and not a request)